File: app/src/main/java/i/am/shiro/amai/fragment/DownloadsFragment.java
package i.am.shiro.amai.fragment;

import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.RecyclerView;

import i.am.shiro.amai.R;
import i.am.shiro.amai.adapter.DownloadJobAdapter;
import i.am.shiro.amai.fragment.dialog.PlaceholderDialogFragment;
import i.am.shiro.amai.viewmodel.DownloadsFragmentModel;

public class DownloadsFragment extends Fragment {

    public DownloadsFragment() {
        super(R.layout.fragment_downloads);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        Toolbar toolbar = view.findViewById(R.id.toolbar);
        toolbar.setOnMenuItemClickListener(this::onActionClick);

        DownloadJobAdapter adapter = new DownloadJobAdapter();

        RecyclerView recyclerView = view.findViewById(R.id.recyclerView);
        recyclerView.setAdapter(adapter);

        ViewModelProviders.of(this)
            .get(DownloadsFragmentModel.class)
            .observeDownloadJobs(this, adapter::submitList);
    }

    private boolean onActionClick(MenuItem menuItem) {
        switch (menuItem.getItemId()) {
            case R.id.action_resume_all:
                showPlaceholder();
                break;
            case R.id.action_pause_all:
                showPlaceholder();
                break;
        }
        return true;
    }

    private void showPlaceholder() {
        PlaceholderDialogFragment dialogFragment = new PlaceholderDialogFragment();
        dialogFragment.show(getChildFragmentManager(), null);
    }
}

File: app/src/main/java/i/am/shiro/amai/fragment/DownloadsFragment.kt
package i.am.shiro.amai.fragment

import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.observe
import i.am.shiro.amai.R
import i.am.shiro.amai.adapter.DownloadJobAdapter
import i.am.shiro.amai.util.amaiViewModels
import i.am.shiro.amai.viewmodel.DownloadsViewModel
import kotlinx.android.synthetic.main.fragment_downloads.*

class DownloadsFragment : Fragment(R.layout.fragment_downloads) {

    private val viewModel by amaiViewModels<DownloadsViewModel>()

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        toolbar.setOnMenuItemClickListener(::onActionClick)

        val adapter = DownloadJobAdapter(
            onDismiss = viewModel::dismiss,
            onCancel = viewModel::cancel,
            onRetry = viewModel::retry,
            onPause = viewModel::pause,
            onResume = viewModel::resume
        )

        recyclerView.adapter = adapter

        viewModel.downloadsLive.observe(viewLifecycleOwner, adapter::submitList)
    }

    private fun onActionClick(menuItem: MenuItem): Boolean {
        when (menuItem.itemId) {
            R.id.action_resume_all -> viewModel.resumeAll()
            R.id.action_pause_all -> viewModel.pauseAll()
        }
        return true
    }
}
